torch/_dynamo/variables/builder.py

```diff
@@ -853,6 +855,42 @@ def wrap_tensor(self, value: torch.Tensor):
 
         return tensor_variable
 
+    def wrap_numpy_ndarray(self, value):
+        assert isinstance(value, np.ndarray)
+        arg_proxy = self.tx.output.root_tracer.create_graph_input(
+            re.sub(r"[^a-zA-Z0-9]+", "_", self.name), type(value)
+        )
+        source = self.get_source()
+        tensor_value = torch.from_numpy(value)
+        options = {"source": source}
+
+        numpy_ndarray_variable = wrap_fx_proxy_cls(
+            target_cls=NumpyNdarrayVariable,
+            tx=self.tx,
+            proxy=self.tx.output.create_proxy(
+                "call_function",
+                torch.from_numpy,
```

Comment:
Isn't this counter to what we agreed on? The FX graph inputs have to take Tensors. So you can't do the `from_numpy` conversion in the graph. It has to happen externally.